File: backend/src/main/java/org/backend/entity/enums/VerificationCodeType.java
package org.backend.entity.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.Getter;

/**
 * 验证码类型枚举
 */
public enum VerificationCodeType {
    LOGIN("login", "登录验证码"),
    REGISTER("register", "注册验证码"),
    FORGET_PASSWORD("forgetPassword", "找回密码验证码");

    private final String value;
    @Getter
    private final String description;

    VerificationCodeType(String value, String description) {
        this.value = value;
        this.description = description;
    }

    /**
     * 根据字符串值获取枚举
     */
    public static VerificationCodeType fromValue(String value) {
        for (VerificationCodeType type : values()) {
            if (type.value.equalsIgnoreCase(value)) {
                return type;
            }
        }
        throw new IllegalArgumentException("无效的验证码类型: " + value);
    }

    @JsonValue
    public String getValue() {
        return value;
    }

}

File: frontend/src/features/auth/types/enums/VerificationCodeType.ts
// 验证码类型枚举 - 定义验证码的不同使用场景
export enum VerificationCodeType {
    // 登录验证码 - 用于用户登录身份验证
    LOGIN = 'login',
    // 注册验证码 - 用于新用户注册时的身份验证
    REGISTER = 'register',
    // 重置密码验证码 - 用于用户重置密码时的身份验证
    RESET_PASSWORD = 'resetPassword',
}